src/domain/core/src/services/pull_service.rs

```diff
@@ -169,6 +169,7 @@ pub enum PullResult {
         new_head: Multihash,
         num_blocks: u64,
         num_records: u64,
+        new_watermark: Option<DateTime<Utc>>,
```

Comment:
While this is going in a direction of making pull results more informative, I'm starting to think that maybe we should do the opposite. 

Perhaps we should have all results only specify `old_head` and `new_head` and let whoever wants - can scan this block range and grab any information they want (number of records, watermark, schema changes etc.)

Reply:
Nice consequence is that this would move "stats gathering" from push-pull ingest / transform / sync to just two places: CLI command output, and flow system.

Downside is the additional metadata reads.

Reply:
Refactored to additional scanning outside `PullService`